### DISCUSSION TITLE
碰到关于 Nginx 反代以及走 Cloudflare tunnel 的问题

### DISCUSSION THREAD

**Discussion by zweie** [AlistGo/alist#5534] [Q&A] [answered]
---
- 系统：Archlinux
- 环境：Podman 容器
  - Alist容器 + Postgres 容器
  - Nginx Proxy Manager 容器对 Alist 进行二级目录的反代 + Cloudflared 容器走 tunnel
- 版本：latest
---
- 搜索以及阅读过文档。
- 目前没有发现类似的 issue 或讨论。
- 不清楚是什么原因导致的。
---

## 问题描述
- 无法添加 Webauthn 。能够进行读取密钥，输入PIN，但在最后网页提示 “Error validating origin”。

### 添加 Webauthn 过程的日志
```
[GIN] 2023/11/17 - 16:36:55 | 200 |     397.657µs |   客户端的IP  | GET      "/alist/api/authn/webauthn_begin_registration"
[GIN] 2023/11/17 - 16:37:04 | 200 |     588.256µs |   客户端的IP  | POST     "/alist/api/authn/webauthn_finish_registration"
```
添加 Webauthn 的过程中真的只有这两行日志。

## 做过的一些检查
- 按照文档中所说检查过 config.json，确认过正确了 `site_url` 。
- 按照文档中所说的设置了 Nginx 的配置，不过有些导致了奇怪的情况，因此最终没有添加进去。
- 删除容器与数据卷并重新创建，在创建容器后只修改了用户名与密码。

### Nginx Proxy Manager 中的设置
```
real_ip_header CF-Connecting-ip;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Real-IP $remote_addr;
```

**Comment by gkl1368** [comment]
我也有这个问题

**Comment by gkl1368** [comment, marked as answer]
> * 系统：Archlinux
> * 环境：Podman 容器
>   
>   * Alist容器 + Postgres 容器
>   * Nginx Proxy Manager 容器对 Alist 进行二级目录的反代 + Cloudflared 容器走 tunnel
> * 版本：latest
> 
> * 搜索以及阅读过文档。
> * 目前没有发现类似的 issue 或讨论。
> * 不清楚是什么原因导致的。
> 
> ## 问题描述
> * 无法添加 Webauthn 。能够进行读取密钥，输入PIN，但在最后网页提示 “Error validating origin”。
> 
> ### 添加 Webauthn 过程的日志
> ```
> [GIN] 2023/11/17 - 16:36:55 | 200 |     397.657µs |   客户端的IP  | GET      "/alist/api/authn/webauthn_begin_registration"
> [GIN] 2023/11/17 - 16:37:04 | 200 |     588.256µs |   客户端的IP  | POST     "/alist/api/authn/webauthn_finish_registration"
> ```
> 
> 添加 Webauthn 的过程中真的只有这两行日志。
> 
> ## 做过的一些检查
> * 按照文档中所说检查过 config.json，确认过正确了 `site_url` 。
> * 按照文档中所说的设置了 Nginx 的配置，不过有些导致了奇怪的情况，因此最终没有添加进去。
> * 删除容器与数据卷并重新创建，在创建容器后只修改了用户名与密码。
> 
> ### Nginx Proxy Manager 中的设置
> ```
> real_ip_header CF-Connecting-ip;
> proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> proxy_set_header X-Forwarded-Proto $scheme;
> proxy_set_header X-Real-IP $remote_addr;
> ```

你好，看我这个issue：https://github.com/alist-org/alist/issues/6294
里面那个测试版已经解决这个问题了
